### ISSUE TITLE
Layout overflow on pages which are fixed to the view port

### ISSUE THREAD

**Issue by seancolsen** [centerofci/mathesar#2150]
Here's a particularly egregious example of the actions pane responding poorly to a narrow viewport.

https://user-images.githubusercontent.com/42411/208987404-3247ddfb-5d08-43a6-90cf-180168b31964.mp4

What's strange here is that when I remove the table description, the actions pane allows itself to get taller before it overflows.

![image](https://user-images.githubusercontent.com/42411/208988319-b8a06ca3-7cd1-483f-9255-86bb6cb1f5f8.png)

Also, something should be done about this problem at a higher level so that individual panes don't push the entire layout to overflow the viewport. Even if the actions pane is responding poorly, it shouldn't take the whole page with it.

## Related

- #2022

**Comment by seancolsen** [comment]
I'm assigning this to @rajatvijay and moving it into the "First Release" milestone because that's how #1990 is tagged and I think we need this ticket to be completed along with it. We'll have this bug as soon as the user is able to make table descriptions. I just bumped into this again, even without grouping I had applied in my previous video. Here's a video that better demonstrates this bug:

https://user-images.githubusercontent.com/42411/210417374-2e5a1732-71b5-4750-a2d7-8353c0bc0ba3.mp4

**Comment by rajatvijay** [comment]
@seancolsen we need to finalize what is the least screen width that we need to support for this. See the below screenshot to find the minimum screen width before it start hiding the table inspector. 
<img width="718" alt="Screenshot 2023-01-05 at 7 19 17 PM" src="https://user-images.githubusercontent.com/11032856/210796054-b05269da-88d1-4bb2-bd70-702d16dd94f3.png">
cc: @kgodey

**Comment by kgodey** [comment]
I'll defer to @seancolsen's judgement here.

**Comment by seancolsen** [comment]
@rajatvijay this ticket isn't about hiding/showing the table inspector. The behavior in your screenshot looks fine to me and seems to indicate that you're not able to reproduce the bug as I experienced it. Notice in my video that the button to show/hide the table inspector is beyond the viewport, as well as the Refresh button. I'm still reproducing this issue in Firefox and Chrome. Perhaps I should be the one to fix this if you're not able to reproduce it.

**Comment by rajatvijay** [comment]
<img width="1428" alt="Screenshot 2023-01-09 at 8 34 06 PM" src="https://user-images.githubusercontent.com/11032856/211339402-98e6437e-19fc-4ebd-a9bc-e4d55a4b3b9d.png">

I have tried this on chrome too, seems to work fine. Assigning this to you @seancolsen for now. If you have some pointers on how to reproduce this, we can get on a call else you can work on this.

**Comment by pavish** [comment]
This issue needs to be handled across all pages which are fixed to the view port. It should be handled at the layout level. This is present in the Data Explorer route as well. I've updated the description to include that.

**Comment by seancolsen** [comment]
I've opened #2249 to close this issue. Here are some notes more broadly which address the scope of this ticket.

- I didn't notice any issues at the layout level, so my fixes in #2249 pertained only to the Table Page.

- @pavish if you notice specific issues with overflow on the Data Explorer, let's open a different ticket for that, with clear steps to reproduce. I wasn't able to spot any overflow issues within the Data Explorer that seemed worth fixing in my PR.

- The problem in the video at the start of this ticket seems to have been fixed in #2201, because after that PR I was no longer able to produce the "particularly egregious" problem. And note that in that video, the site header is still re-sizing, which tells me that the layout is probably fine.
